resources/prosody-plugins/mod_auth_token.lua

```diff
@@ -31,16 +31,29 @@ function init_session(event)
     local session, request = event.session, event.request;
     local query = request.url.query;
 
-    if query ~= nil then
-        local params = formdecode(query);
+    local token = nil;
 
-        -- The following fields are filled in the session, by extracting them
-        -- from the query and no validation is being done.
-        -- After validating auth_token will be cleaned in case of error and few
-        -- other fields will be extracted from the token and set in the session
+    -- preferentially extract token from Authorization header
+    if request.headers["authorization"] and starts_with(request.headers["authorization"],'Bearer ') then
```

Comment:
I'm woried about the starts_with performance here, checking it for every request ... 
Can we control the header name in the nginx that fronts it, so we don't need to be guessing?

Reply:
Fair worry, removing the second part of the check.